elide-core/src/main/java/com/yahoo/elide/ElideSettingsBuilder.java

```diff
@@ -125,6 +128,11 @@ public ElideSettingsBuilder withEntityDictionary(EntityDictionary entityDictiona
         return this;
     }
 
+    public ElideSettingsBuilder withNonEntityDictionary(NonEntityDictionary nonEntityDictionary) {
```

Comment:
Another possibility here is to embed NonEntityDictionary inside EntityDictionary itself - and add some new methods in EntityDictionary that allow lookups of entities & non-entities and their fields alike. 

Then those methods can be used in contexts where we don't care if the entity is an elide model or not (like path traversal).

I'm not sure what will work better in practice so I would try both and see which one is cleaner.

Reply:
I like this idea. let me try this approach as well.